x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -257,7 +315,14 @@ export const SearchBar: FC<SearchBarProps> = ({
         value: searchValue,
         onInput: (e: React.UIEvent<HTMLInputElement>) => setSearchValue(e.currentTarget.value),
         'data-test-subj': 'nav-search-input',
-        inputRef: setSearchRef,
+        inputRef: (input) => {
+          setSearchRef(input);
+          if (chromeStyle === 'project' && input) {
+            // while the input ref is in flight, we set focus on it
+            // to autofocus the input when it appears
+            input.focus();
```

Comment:
Would the `autoFocus` prop work here? 

```
searchProps={{
        autoFocus: chromeStyle === 'project',
```

Reply:
Thanks, this will be much better than my hack 🤦🏻

Reply:
f4678a1ebaa13e90d82808baba0a9dbfd001db6a